datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/tag/TagType.java

```diff
@@ -59,6 +89,64 @@ public EntityType type() {
         }
     }
 
+    @Override
+    public SearchResults search(@Nonnull String query,
+                                @Nullable List<FacetFilterInput> filters,
+                                int start,
+                                int count,
+                                @Nonnull QueryContext context) throws Exception {
+        final Map<String, String> facetFilters = ResolverUtils.buildFacetFilters(filters, TAG_SEARCH_CONFIG.getFacetFields());
+        final CollectionResponse<com.linkedin.tag.Tag> searchResult = _tagClient.search(query, null, facetFilters, null, start, count);
+        return SearchResultsMapper.map(searchResult, TagMapper::map);
+    }
+
+    @Override
+    public Tag update(@Nonnull TagUpdate input, @Nonnull QueryContext context) throws Exception {
+        // TODO: Verify that updater is owner.
+        final CorpuserUrn actor = new CorpuserUrn(context.getActor());
+        final com.linkedin.tag.Tag partialTag = TagUpdateMapper.map(input);
+
+        // Create Audit Stamp
+        final AuditStamp auditStamp = new AuditStamp();
+        auditStamp.setActor(actor, SetMode.IGNORE_NULL);
+        auditStamp.setTime(System.currentTimeMillis());
+
+        if (partialTag.hasOwnership()) {
+            partialTag.getOwnership().setLastModified(auditStamp);
+        } else {
+            final Ownership ownership = new Ownership();
```

Comment:
Seems like this ownership creation logic will probably be duplicated elsewhere (or already is)... we can consider extracting it

Reply:
good idea- this can be done in a followup